tests/run/i5350.scala

```diff
@@ -0,0 +1,10 @@
+object Test {
+  def foo = {
+    lazy val bar: Unit = println("Hello")
```

Comment:
@Blaisorblade does it make sense for this `lazy val` to be stable?

Reply:
Yes. The definition of `isStable` doesn't check for `Lazy`, so `Symbol.isStable` should only imply idempotence.

To detect if a value is pure, Realizability checks `sym.isStable && !sym.is(Lazy | Erased, butNot = Module)`. Maybe here you should check
`sym.isStable && !sym.is(Lazy, butNot = Module)`? But on modules, this overlaps with #2266 and #4765.

That all needs testcases and consideration?

Reply:
Actually modules do not reach [this line](https://github.com/lampepfl/dotty/pull/5703/files#diff-60732058b4c21c7127418de34cb149e9R403).  Hence the current condition does not affect the module init elimination.